### ISSUE TITLE
MLKit selfie segmentation increases the size of android app

### ISSUE THREAD

**Issue by A-KVerma** [googlesamples/mlkit#386]
I am using selfie segmentation in my android app, the app size has increased from 50 MB to 110 MB due to a library named `libxeno_native.so` which is taking almost 50MB. Any idea to reduce the app size?

**Comment by steven-chengji-yan** [comment]
Hi A-Kverma,

Which version of selfie-segmentation are you using? I just downloaded the latest related aar from GMaven. 
https://maven.google.com/web/index.html?q=mlkit#com.google.mlkit:segmentation-selfie:16.0.0-beta2
https://maven.google.com/web/index.html?q=mlkit#com.google.mlkit:mediapipe-internal:17.0.0-beta4
Adding them together, the size is about 17MB. And **that is for 4 ABIs**. If you use [Android App Bundle](https://developer.android.com/guide/app-bundle) as required by Google Play, one end-user will only need to download the libxeno_native shared library for 1 ABI, which will decrease the size more.

**Comment by steven-chengji-yan** [comment]
Feel free to re-open if the previous reply doesn't answer your question.

**Comment by A-KVerma** [comment]
Yes 50MB means adding all the 4 ABIs together, the size is between (15-20)MB for each ABIs. 
Got it thanks for your reply, are you guys planning to further reduce the shared library libxeno_native size in near future ?

**Comment by steven-chengji-yan** [comment]
Replied in https://github.com/googlesamples/mlkit/issues/387
